components/server/src/auth/host-context-provider-impl.ts

```diff
@@ -82,16 +82,19 @@ export class HostContextProviderImpl implements HostContextProvider {
             const { host } = config;
 
             const existingContext = this.dynamicHosts.get(host);
-            const existingConfig = existingContext && existingContext.authProvider.params;
-            if (existingConfig && config.id === existingConfig.id) {
-                if (existingConfig.host !== config.host) {
+            if (existingContext) {
+                const existingConfig = existingContext.authProvider.params;
+                const sameId = config.id === existingConfig.id;
+                const sameHost = config.host === existingConfig.host;
+                if (!sameHost || !sameId) {
```

Comment:
Idee that we now treat "different IDs" to be a broken state. Correct? Before we would treat them as different provider.

Reply:
Right, that wasn't though through. Indeed it could happen, that one provider is removed and another (legit) was created in the same interval. 

Going to revert that.